### PR TITLE
use Model Type for TorchHub Registry

### PR DESCRIPTION
**Pull request by da-tubi**
+ Deprecate pre/post/OUTPUT_SCHEMA support for TorchHub Registry
+ add Model Type support for TorchHub Registry

### CODE REVIEW

File: experimental/torchhub/setup.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 setup(
     name="rikai-torchhub",
-    version="0.0.2",
+    version="0.1.0",
```
Comment (darcy-shen) [inline experimental/torchhub/setup.py:5]: to distinguish with 0.1.x

File: experimental/torchhub/rikai/experimental/torchhub/torchhub_registry.py [hunk 2]
```diff
@@ -24,41 +24,30 @@
 
 
 class TorchHubModelSpec(ModelSpec):
-    def __init__(self, repo_or_dir: str, model: str, raw_spec: ModelSpec):
+    def __init__(self, repo_or_dir: str, model: str, raw_spec: "ModelSpec"):
```
Comment (darcy-shen) [inline experimental/torchhub/rikai/experimental/torchhub/torchhub_registry.py:27]: raw_spec is from Scala ModelSpec, that's why double quotes are used here.

File: python/rikai/spark/sql/codegen/pytorch.py [hunk 3]
```diff
@@ -41,7 +41,7 @@ def move_tensor_to_device(data, device):
     return data
 
 
-def generate_udf(payload: ModelSpec):
+def generate(payload: ModelSpec, is_udf: bool = True):
```
Comment (da-tubi) [inline python/rikai/spark/sql/codegen/pytorch.py:44]: Use `is_udf = False` to generate the function which can be used without PySpark

File: python/rikai/contrib/torchhub/pytorch/vision/resnet.py [hunk 4]
```diff
@@ -12,41 +12,58 @@
...
+            self.model is not None
+        ), "model has not been initialized via load_model"
+        batch = self.model(images)
         results = []
         for result in batch:
             results.append(result.detach().cpu().tolist())
         return results
 
-    return post_process_func
+    def release(self):
```
Comment (darcy-shen) [inline python/rikai/contrib/torchhub/pytorch/vision/resnet.py:63]: @eddyxu can we move it to TorchModelType?

File: experimental/torchhub/tests/test_resnet.py [hunk 5]
```diff
@@ -13,18 +13,43 @@
...
+)
+from rikai.spark.sql.codegen.base import func_from_spec
+
+
+version = f"v{torchvision.__version__.split('+', maxsplit=1)[0]}"
+work_dir = Path().absolute().parent.parent
+image_path = f"{work_dir}/python/tests/assets/test_image.jpg"
+
+
+def test_resnet_model_type():
```
Comment (darcy-shen) [inline experimental/torchhub/tests/test_resnet.py:34]: This is an attempt to provide mode type utilities. It aims to provide a pyspark-free way to test the model type.

File: experimental/torchhub/rikai/experimental/torchhub/torchhub_registry.py [hunk 6]
```diff
@@ -24,41 +24,30 @@
...
+        # remove None value
+        if not spec["schema"]:
+            del spec["schema"]
         # defaults to the `pytorch` flavor
         if not spec["model"]["flavor"]:
             spec["model"]["flavor"] = "pytorch"
 
+        repo_proj = repo_or_dir.split(":")[0].replace("/", ".")
+        if not spec["model"]["type"]:
+            model_type = f"rikai.contrib.torchhub.{repo_proj}.{model}"
```
Comment (darcy-shen) [inline experimental/torchhub/rikai/experimental/torchhub/torchhub_registry.py:47]: extra convention to find model type for torchhub model

File: experimental/torchhub/rikai/experimental/torchhub/torchhub_registry.py [hunk 7]
```diff
@@ -24,41 +24,30 @@
...
 class TorchHubModelSpec(ModelSpec):
-    def __init__(self, repo_or_dir: str, model: str, raw_spec: ModelSpec):
+    def __init__(self, repo_or_dir: str, model: str, raw_spec: "ModelSpec"):
         spec = {
             "version": "1.0",
             "schema": raw_spec["schema"],
-            "model": {"flavor": raw_spec["flavor"], "uri": raw_spec["uri"]},
-            "transforms": {
-                "pre": raw_spec.get("preprocessor", None),
-                "post": raw_spec.get("postprocessor", None),
```
Comment (darcy-shen) [inline experimental/torchhub/rikai/experimental/torchhub/torchhub_registry.py:34]: the pre/post processor approach is removed here

**Review by Renkai — APPROVED** [review]
LGTM

**Pull request merged**